### DISCUSSION TITLE
animated css appears at first render although opacity is 0

### DISCUSSION THREAD

**Discussion by esrefbozkurt** [neuefische/questions#412] [Web]
Hey,
I styled a search bar which ist animated and expands on click (used an template and styled it for my usage). The opacity of the hidden input field is 0. At the first rendering of the website, it appears for a short moment. It looks like it is a mix of the default input field with no styling an my styled input field. pls help

[PR](https://github.com/esrefbozkurt/capstone-project/pull/12)

**Comment by ufelgen** [comment]
Hey Eshi, I think found a solution but I'm not sure if it's acceptable and/or elegant. I recommend asking one of the coaches on Monday... 

First I identified the problem as the transition property on your input element and removed it. 

Next, I defined it inside a separate class (inside the StyledSearchBar component):
`& .transition {
    transition: all 0.5s;
  }`

Then, I added a function to add this class to the search bar again:
` function addTransition() {
    if (typeof window !== "undefined") {
      document.getElementById("search-bar").classList.add("transition");
    }
  }`

Finally, this function needs to be invoked, but not immediately on page render (as this recreates the same problem as before). Here I used setTimeout to call the function 20ms after page render: 
`  setTimeout(() => {
    if (typeof window !== "undefined") {
      window.onload = addTransition();
    }
  }, 20);`

again, not sure if this is an acceptable solution, but maybe it helps.

**Comment by actyralf** [comment]
Maybe you experience these issues because styled components are not set up for server side rendering correctly.
In our course we recommended to add a custom `_document.js` to your `/pages` directory as described here: https://github.com/neuefische/ffm-web-22-1/blob/main/docs/next-js-styled-components-ssr-fix.md
I'm not sure if this works with your capstone template, but you could give it a try.
